### ISSUE TITLE
composition with other micro modules

### ISSUE THREAD

**Issue by saggiyogesh** [possibilities/micro-api#23]
Seems like the handler is breaking composition with micro modules like 
https://github.com/kandros/micro-jwt-auth
so something like this is not working 
```
const microApi = require('micro-api');
const jwtAuth = require('micro-jwt-auth');

const handler = function (req, res) {
  return 'Hello micro';
};

module.exports = microApi([{
  method: 'get',
  path: '/tt',
  handler: jwtAuth('my_jwt_secret')(handler)
}]);
```

as `jwtAuth` requires `req` & `res` as arguments but are passed in object https://github.com/possibilities/micro-api/blob/master/src/index.js#L70

Your thoughts, please.

**Comment by possibilities** [comment]
Yeah I guess this means individual routes break the contract. I can't think
of any nice way to paper over this. The main idea of this library was to
provide a shorthand for marrying routes to micro handlers... I'll give it
some more thought, I don't even mind breaking micro-apis api completely we
can accomplish the original goal without breaking composibility.

**Comment by saggiyogesh** [comment]
I wrote a compose function to handle this
```
const argsComposer = fn => async ({ req, res }) => {
  return await fn(req, res);
};
```
This will handle the object from microApi & pass `req`, `res` to next compose function.
